src/main/java/com/spotify/docker/client/CompressedDirectory.java

```diff
@@ -265,6 +265,27 @@ private Visitor(final Path root, ImmutableList<DockerIgnorePathMatcher> ignoreMa
     }
 
     @Override
+    public FileVisitResult preVisitDirectory(Path dir,
+                                             BasicFileAttributes attrs) throws IOException {
+      if (Files.isSameFile(dir, root)) {
+        return FileVisitResult.CONTINUE;
+      }
+
+      final Path relativePath = root.relativize(dir);
+
+      if (exclude(ignoreMatchers, relativePath)) {
+        return FileVisitResult.CONTINUE;
+      }
+
+      final TarArchiveEntry entry = new TarArchiveEntry(dir.toFile());
+      entry.setName(relativePath.toString());
+      entry.setMode(getFileMode(dir));
```

Comment:
I'm not that familiar with `TarArchiveEntry`, but I guess we don't need `entry.setSize()`?